crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -369,6 +369,32 @@ where
         Ok(())
     }
 
+    /// Schedule a state change that restarts the active state.
+    /// This will fail if there is a scheduled operation
+    pub fn restart(&mut self) -> Result<(), StateError> {
+        if self.scheduled.is_some() {
+            return Err(StateError::StateAlreadyQueued);
+        }
+
+        if let Some(state) = self.stack.last() {
```

Comment:
These if let s can be replaced with unwraps, as the stack is defined as always having at least one state when not in a transition (the one that's currently being executed)

Reply:
Gotcha! 

I can add that bit of info as a doc-comment on the `stack` field as well, since it could be nice for future contributors to know.